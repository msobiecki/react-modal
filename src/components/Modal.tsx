import {
  useReducer,
  useEffect,
  forwardRef,
  useImperativeHandle,
  useRef,
  useCallback,
} from "react";
import PropTypes, { InferProps } from "prop-types";

import {
  reducer as modalReducer,
  initialValues as modalInitialValues,
} from "./Modal.reducer";

import { Wrapper, Overlay, Panel, Close, CloseIcon } from "./Modal.styled";

import CLOSE_ICON from "../assets/images/Close.svg";

type InferPropType<PropTypes, DefaultProps, Props = InferProps<PropTypes>> = {
  [Key in keyof Props]: Key extends keyof DefaultProps
    ? Props[Key] | DefaultProps[Key]
    : Props[Key];
};

const modalPropTypes = {
  children: PropTypes.node,
  options: PropTypes.shape({
    openingDuration: PropTypes.number,
    closingDuration: PropTypes.number,
    zIndex: PropTypes.number,
    isOverlay: PropTypes.bool,
    isOverlayClosing: PropTypes.bool,
    overlayPadding: PropTypes.string,
    overlayBackground: PropTypes.string,
    isPanelCloseButton: PropTypes.bool,
    isPanelRestrictedSize: PropTypes.bool,
    panelRestrictedWidth: PropTypes.string,
    panelRestrictedHeight: PropTypes.string,
    panelWidth: PropTypes.string,
    panelHeight: PropTypes.string,
    panelBorderRadius: PropTypes.string,
    panelBorder: PropTypes.string,
    panelBackground: PropTypes.string,
    panelBoxShadow: PropTypes.string,
    panelPadding: PropTypes.string,
  }),
  onOpenStart: PropTypes.func,
  onOpenEnd: PropTypes.func,
  onCloseStart: PropTypes.func,
  onCloseEnd: PropTypes.func,
};

const modalDefaultProps = {
  options: {
    openingDuration: modalInitialValues.openingDuration,
    closingDuration: modalInitialValues.closingDuration,
    zIndex: modalInitialValues.zIndex,
    isOverlay: modalInitialValues.isOverlay,
    isOverlayClosing: modalInitialValues.isOverlayClosing,
    overlayPadding: modalInitialValues.overlayPadding,
    overlayBackground: modalInitialValues.overlayBackground,
    isPanelCloseButton: modalInitialValues.isPanelCloseButton,
    isPanelRestrictedSize: modalInitialValues.isPanelRestrictedSize,
    panelRestrictedWidth: modalInitialValues.panelRestrictedWidth,
    panelRestrictedHeight: modalInitialValues.panelRestrictedHeight,
    panelWidth: modalInitialValues.panelWidth,
    panelHeight: modalInitialValues.panelHeight,
    panelBorderRadius: modalInitialValues.panelBorderRadius,
    panelBorder: modalInitialValues.panelBorder,
    panelBackground: modalInitialValues.panelBackground,
    panelBoxShadow: modalInitialValues.panelBoxShadow,
    panelPadding: modalInitialValues.panelPadding,
  },
  onOpenStart: () => null,
  onOpenEnd: () => null,
  onCloseStart: () => null,
  onCloseEnd: () => null,
};

type ModalPropsType = InferPropType<
  typeof modalPropTypes,
  typeof modalDefaultProps
>;

type ForwardedRefType = {
  openModal: () => void;
  closeModal: () => void;
};

export const Modal = forwardRef<ForwardedRefType, ModalPropsType>(
  (
    {
      children,
      options,
      onOpenStart,
      onOpenEnd,
      onCloseStart,
      onCloseEnd,
    }: ModalPropsType,
    forwardedRef
  ) => {
    const panelRef = useRef<HTMLDivElement>(null);
    const [modalState, modalDispatch] = useReducer(
      modalReducer,
      Object.assign({}, modalInitialValues, options)
    );

    const handleActionOnOpenStart = useCallback(
      () => typeof onOpenStart === "function" && onOpenStart(),
      [onOpenStart]
    );
    const handleActionOnOpenEnd = useCallback(
      () => typeof onOpenEnd === "function" && onOpenEnd(),
      [onOpenEnd]
    );
    const handleActionOnCloseStart = useCallback(
      () => typeof onCloseStart === "function" && onCloseStart(),
      [onCloseStart]
    );
    const handleActionOnCloseEnd = useCallback(
      () => typeof onCloseEnd === "function" && onCloseEnd(),
      [onCloseEnd]
    );

    const openModal = useCallback(() => {
      if (modalState.isClosed) modalDispatch({ type: "OPEN_START" });
    }, [modalState.isClosed]);

    const closeModal = useCallback(() => {
      if (modalState.isOpened) modalDispatch({ type: "CLOSE_START" });
    }, [modalState.isOpened]);

    const handleClickDocumentEventListener = useCallback(
      (event: MouseEvent) => {
        if (
          !(
            panelRef &&
            panelRef.current &&
            panelRef.current.contains(event.target as Element)
          )
        ) {
          closeModal();
        }
      },
      [closeModal]
    );

    useEffect(() => {
      let timeout: ReturnType<typeof setTimeout>;
      if (modalState.isOpening) {
        handleActionOnOpenStart();
        timeout = setTimeout(() => {
          modalDispatch({ type: "OPEN_END" });
          handleActionOnOpenEnd();
        }, modalState.openingDuration);
      }
      return () => {
        if (timeout) clearTimeout(timeout);
      };
    }, [
      handleActionOnOpenEnd,
      handleActionOnOpenStart,
      modalState.isOpening,
      modalState.openingDuration,
    ]);

    useEffect(() => {
      let timeout: ReturnType<typeof setTimeout>;
      if (modalState.isClosing) {
        handleActionOnCloseStart();
        timeout = setTimeout(() => {
          modalDispatch({ type: "CLOSE_END" });
          handleActionOnCloseEnd();
        }, modalState.closingDuration);
      }

      return () => {
        if (timeout) clearTimeout(timeout);
      };
    }, [
      handleActionOnCloseEnd,
      handleActionOnCloseStart,
      modalState.closingDuration,
      modalState.isClosing,
    ]);

    useEffect(() => {
      if (modalState.isOverlay && modalState.isOverlayClosing)
        document.addEventListener("click", handleClickDocumentEventListener);
      return () => {
        if (modalState.isOverlay && modalState.isOverlayClosing)
          document.removeEventListener(
            "click",
            handleClickDocumentEventListener
          );
      };
    }, [
      modalState.isOpening,
      modalState.isOpened,
      modalState.isOverlay,
      modalState.isOverlayClosing,
      handleClickDocumentEventListener,
    ]);

    useImperativeHandle(forwardedRef, () => ({
      openModal: () => openModal(),
      closeModal: () => closeModal(),
    }));

    const renderPanel = () => {
      return (
        <Panel
          ref={panelRef}
          isVisible={
            modalState.isOpening || modalState.isOpened || modalState.isClosing
              ? true
              : false
          }
          isOpening={modalState.isOpening ? true : false}
          isClosing={modalState.isClosing ? true : false}
          openingDuration={modalState.openingDuration}
          closingDuration={modalState.closingDuration}
          padding={modalState.panelPadding}
          isRestrictedSize={modalState.isPanelRestrictedSize}
          restrictedWidth={modalState.panelRestrictedWidth}
          restrictedHeight={modalState.panelRestrictedHeight}
          width={modalState.panelWidth}
          height={modalState.panelHeight}
          borderRadius={modalState.panelBorderRadius}
          border={modalState.panelBorder}
          background={modalState.panelBackground}
          boxShadow={modalState.panelBoxShadow}
        >
          {modalState.isPanelCloseButton && (
            <Close
              onClick={() => {
                closeModal();
              }}
            >
              <CloseIcon src={CLOSE_ICON} />
            </Close>
          )}

          {children}
        </Panel>
      );
    };

    return (
      <>
        {!modalState.isClosed ? (
          <Wrapper
            isOverlay={modalState.isOverlay ? true : false}
            zIndex={modalState.zIndex}
          >
            {modalState.isOverlay ? (
              <Overlay
                isVisible={
                  modalState.isOpening ||
                  modalState.isOpened ||
                  modalState.isClosing
                    ? true
                    : false
                }
                isOpening={modalState.isOpening ? true : false}
                isClosing={modalState.isClosing ? true : false}
                openingDuration={modalState.openingDuration}
                closingDuration={modalState.closingDuration}
                padding={modalState.overlayPadding}
                background={modalState.overlayBackground}
              >
                {renderPanel()}
              </Overlay>
            ) : (
              <>{renderPanel()}</>
            )}
          </Wrapper>
        ) : (
          ""
        )}
      </>
    );
  }
);
Modal.displayName = "Modal";
Modal.defaultProps = modalDefaultProps;

export default Modal;
