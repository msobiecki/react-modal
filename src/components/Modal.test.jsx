import * as React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import { Modal } from "./Modal";

describe("[MODAL]", () => {
  it("[1] Initial", () => {
    const { asFragment } = render(<Modal />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("[2] Custom duration", () => {
    const { asFragment } = render(
      <Modal
        options={{
          openingDuration: 5000,
          closingDuration: 3000,
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[3] Custom z index", () => {
    const { asFragment } = render(
      <Modal
        options={{
          zIndex: 10,
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[4] With overlay", () => {
    const { asFragment } = render(
      <Modal
        options={{
          isOverlay: true,
          openingDuration: 800,
          closingDuration: 500,
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[5] With custom overlay", () => {
    const { asFragment } = render(
      <Modal
        options={{
          isOverlay: true,
          overlayPadding: "20px 10px 30px 40px",
          overlayBackground: "rgba(111, 222, 121, 0.8)",
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[6] With prevent close overlay", () => {
    const { asFragment } = render(
      <Modal
        options={{
          isOverlay: true,
          isOverlayClosing: false,
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[7] With custom panel", () => {
    const { asFragment } = render(
      <Modal
        options={{
          panelBorderRadius: "20px 30px 10px 40px",
          panelBorder: "5px solid #332321",
          panelBackground: "rgba(155,155,155,1)",
          panelBoxShadow: "0 0 5px 0 rgba(0, 0, 0, .2)",
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[8] With restricted children width", () => {
    const { asFragment } = render(
      <Modal>
        <div style={{ width: "100%", maxWidth: "640px" }}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum
          pretium orci eu semper blandit. Fusce bibendum odio eu dolor
          elementum, accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris
          lobortis semper. Phasellus ut libero quis quam ornare tincidunt quis
          id diam. Pellentesque luctus libero ut dignissim iaculis. Quisque nec
          orci ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus
          enim, eget maximus lorem. Vestibulum ante ipsum primis in faucibus
          orci luctus et ultrices posuere cubilia curae; Pellentesque varius
          eget enim at varius. Lorem ipsum dolor sit amet, consectetur
          adipiscing elit. Vestibulum pretium orci eu semper blandit. Fusce
          bibendum odio eu dolor elementum, accumsan rhoncus mauris luctus. Sed
          feugiat iaculis mauris lobortis semper. Phasellus ut libero quis quam
          ornare tincidunt quis id diam. Pellentesque luctus libero ut dignissim
          iaculis. Quisque nec orci ullamcorper, pretium quam eu, gravida ipsum.
          Morbi at faucibus enim, eget maximus lorem. Vestibulum ante ipsum
          primis in faucibus orci luctus et ultrices posuere cubilia curae;
          Pellentesque varius eget enim at varius. Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Vestibulum pretium orci eu semper
          blandit. Fusce bibendum odio eu dolor elementum, accumsan rhoncus
          mauris luctus. Sed feugiat iaculis mauris lobortis semper. Phasellus
          ut libero quis quam ornare tincidunt quis id diam. Pellentesque luctus
          libero ut dignissim iaculis. Quisque nec orci ullamcorper, pretium
          quam eu, gravida ipsum. Morbi at faucibus enim, eget maximus lorem.
          Vestibulum ante ipsum primis in faucibus orci luctus et ultrices
          posuere cubilia curae; Pellentesque varius eget enim at varius. Lorem
          ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum pretium
          orci eu semper blandit. Fusce bibendum odio eu dolor elementum,
          accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris lobortis
          semper. Phasellus ut libero quis quam ornare tincidunt quis id diam.
          Pellentesque luctus libero ut dignissim iaculis. Quisque nec orci
          ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus enim,
          eget maximus lorem. Vestibulum ante ipsum primis in faucibus orci
          luctus et ultrices posuere cubilia curae; Pellentesque varius eget
          enim at varius.
        </div>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[9] With overlay with restricted children width", () => {
    const { asFragment } = render(
      <Modal
        options={{
          isOverlay: true,
        }}
      >
        <div style={{ width: "100%", maxWidth: "640px" }}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum
          pretium orci eu semper blandit. Fusce bibendum odio eu dolor
          elementum, accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris
          lobortis semper. Phasellus ut libero quis quam ornare tincidunt quis
          id diam. Pellentesque luctus libero ut dignissim iaculis. Quisque nec
          orci ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus
          enim, eget maximus lorem. Vestibulum ante ipsum primis in faucibus
          orci luctus et ultrices posuere cubilia curae; Pellentesque varius
          eget enim at varius. Lorem ipsum dolor sit amet, consectetur
          adipiscing elit. Vestibulum pretium orci eu semper blandit. Fusce
          bibendum odio eu dolor elementum, accumsan rhoncus mauris luctus. Sed
          feugiat iaculis mauris lobortis semper. Phasellus ut libero quis quam
          ornare tincidunt quis id diam. Pellentesque luctus libero ut dignissim
          iaculis. Quisque nec orci ullamcorper, pretium quam eu, gravida ipsum.
          Morbi at faucibus enim, eget maximus lorem. Vestibulum ante ipsum
          primis in faucibus orci luctus et ultrices posuere cubilia curae;
          Pellentesque varius eget enim at varius. Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Vestibulum pretium orci eu semper
          blandit. Fusce bibendum odio eu dolor elementum, accumsan rhoncus
          mauris luctus. Sed feugiat iaculis mauris lobortis semper. Phasellus
          ut libero quis quam ornare tincidunt quis id diam. Pellentesque luctus
          libero ut dignissim iaculis. Quisque nec orci ullamcorper, pretium
          quam eu, gravida ipsum. Morbi at faucibus enim, eget maximus lorem.
          Vestibulum ante ipsum primis in faucibus orci luctus et ultrices
          posuere cubilia curae; Pellentesque varius eget enim at varius. Lorem
          ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum pretium
          orci eu semper blandit. Fusce bibendum odio eu dolor elementum,
          accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris lobortis
          semper. Phasellus ut libero quis quam ornare tincidunt quis id diam.
          Pellentesque luctus libero ut dignissim iaculis. Quisque nec orci
          ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus enim,
          eget maximus lorem. Vestibulum ante ipsum primis in faucibus orci
          luctus et ultrices posuere cubilia curae; Pellentesque varius eget
          enim at varius.
        </div>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[10] With restricted children width with restricted panel size", () => {
    const { asFragment } = render(
      <Modal
        options={{
          isPanelRestrictedSize: true,
        }}
      >
        <div style={{ width: "100%", maxWidth: "640px" }}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum
          pretium orci eu semper blandit. Fusce bibendum odio eu dolor
          elementum, accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris
          lobortis semper. Phasellus ut libero quis quam ornare tincidunt quis
          id diam. Pellentesque luctus libero ut dignissim iaculis. Quisque nec
          orci ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus
          enim, eget maximus lorem. Vestibulum ante ipsum primis in faucibus
          orci luctus et ultrices posuere cubilia curae; Pellentesque varius
          eget enim at varius. Lorem ipsum dolor sit amet, consectetur
          adipiscing elit. Vestibulum pretium orci eu semper blandit. Fusce
          bibendum odio eu dolor elementum, accumsan rhoncus mauris luctus. Sed
          feugiat iaculis mauris lobortis semper. Phasellus ut libero quis quam
          ornare tincidunt quis id diam. Pellentesque luctus libero ut dignissim
          iaculis. Quisque nec orci ullamcorper, pretium quam eu, gravida ipsum.
          Morbi at faucibus enim, eget maximus lorem. Vestibulum ante ipsum
          primis in faucibus orci luctus et ultrices posuere cubilia curae;
          Pellentesque varius eget enim at varius. Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Vestibulum pretium orci eu semper
          blandit. Fusce bibendum odio eu dolor elementum, accumsan rhoncus
          mauris luctus. Sed feugiat iaculis mauris lobortis semper. Phasellus
          ut libero quis quam ornare tincidunt quis id diam. Pellentesque luctus
          libero ut dignissim iaculis. Quisque nec orci ullamcorper, pretium
          quam eu, gravida ipsum. Morbi at faucibus enim, eget maximus lorem.
          Vestibulum ante ipsum primis in faucibus orci luctus et ultrices
          posuere cubilia curae; Pellentesque varius eget enim at varius. Lorem
          ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum pretium
          orci eu semper blandit. Fusce bibendum odio eu dolor elementum,
          accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris lobortis
          semper. Phasellus ut libero quis quam ornare tincidunt quis id diam.
          Pellentesque luctus libero ut dignissim iaculis. Quisque nec orci
          ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus enim,
          eget maximus lorem. Vestibulum ante ipsum primis in faucibus orci
          luctus et ultrices posuere cubilia curae; Pellentesque varius eget
          enim at varius.
        </div>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[11] With overlay with restricted children width with restricted panel size", () => {
    const { asFragment } = render(
      <Modal
        options={{
          isOverlay: true,
          isPanelRestrictedSize: true,
        }}
      >
        <div style={{ width: "100%", maxWidth: "640px" }}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum
          pretium orci eu semper blandit. Fusce bibendum odio eu dolor
          elementum, accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris
          lobortis semper. Phasellus ut libero quis quam ornare tincidunt quis
          id diam. Pellentesque luctus libero ut dignissim iaculis. Quisque nec
          orci ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus
          enim, eget maximus lorem. Vestibulum ante ipsum primis in faucibus
          orci luctus et ultrices posuere cubilia curae; Pellentesque varius
          eget enim at varius. Lorem ipsum dolor sit amet, consectetur
          adipiscing elit. Vestibulum pretium orci eu semper blandit. Fusce
          bibendum odio eu dolor elementum, accumsan rhoncus mauris luctus. Sed
          feugiat iaculis mauris lobortis semper. Phasellus ut libero quis quam
          ornare tincidunt quis id diam. Pellentesque luctus libero ut dignissim
          iaculis. Quisque nec orci ullamcorper, pretium quam eu, gravida ipsum.
          Morbi at faucibus enim, eget maximus lorem. Vestibulum ante ipsum
          primis in faucibus orci luctus et ultrices posuere cubilia curae;
          Pellentesque varius eget enim at varius. Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Vestibulum pretium orci eu semper
          blandit. Fusce bibendum odio eu dolor elementum, accumsan rhoncus
          mauris luctus. Sed feugiat iaculis mauris lobortis semper. Phasellus
          ut libero quis quam ornare tincidunt quis id diam. Pellentesque luctus
          libero ut dignissim iaculis. Quisque nec orci ullamcorper, pretium
          quam eu, gravida ipsum. Morbi at faucibus enim, eget maximus lorem.
          Vestibulum ante ipsum primis in faucibus orci luctus et ultrices
          posuere cubilia curae; Pellentesque varius eget enim at varius. Lorem
          ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum pretium
          orci eu semper blandit. Fusce bibendum odio eu dolor elementum,
          accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris lobortis
          semper. Phasellus ut libero quis quam ornare tincidunt quis id diam.
          Pellentesque luctus libero ut dignissim iaculis. Quisque nec orci
          ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus enim,
          eget maximus lorem. Vestibulum ante ipsum primis in faucibus orci
          luctus et ultrices posuere cubilia curae; Pellentesque varius eget
          enim at varius.
        </div>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[12] With fullscreen panel width and height with restricted panel size", () => {
    const { asFragment } = render(
      <Modal
        options={{
          isPanelRestrictedSize: true,
          panelWidth: "100%",
          panelHeight: "100%",
        }}
      >
        <div>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum
          pretium orci eu semper blandit. Fusce bibendum odio eu dolor
          elementum, accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris
          lobortis semper. Phasellus ut libero quis quam ornare tincidunt quis
          id diam. Pellentesque luctus libero ut dignissim iaculis. Quisque nec
          orci ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus
          enim, eget maximus lorem. Vestibulum ante ipsum primis in faucibus
          orci luctus et ultrices posuere cubilia curae; Pellentesque varius
          eget enim at varius. Lorem ipsum dolor sit amet, consectetur
          adipiscing elit. Vestibulum pretium orci eu semper blandit. Fusce
          bibendum odio eu dolor elementum, accumsan rhoncus mauris luctus. Sed
          feugiat iaculis mauris lobortis semper. Phasellus ut libero quis quam
          ornare tincidunt quis id diam. Pellentesque luctus libero ut dignissim
          iaculis. Quisque nec orci ullamcorper, pretium quam eu, gravida ipsum.
          Morbi at faucibus enim, eget maximus lorem. Vestibulum ante ipsum
          primis in faucibus orci luctus et ultrices posuere cubilia curae;
          Pellentesque varius eget enim at varius. Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Vestibulum pretium orci eu semper
          blandit. Fusce bibendum odio eu dolor elementum, accumsan rhoncus
          mauris luctus. Sed feugiat iaculis mauris lobortis semper. Phasellus
          ut libero quis quam ornare tincidunt quis id diam. Pellentesque luctus
          libero ut dignissim iaculis. Quisque nec orci ullamcorper, pretium
          quam eu, gravida ipsum. Morbi at faucibus enim, eget maximus lorem.
          Vestibulum ante ipsum primis in faucibus orci luctus et ultrices
          posuere cubilia curae; Pellentesque varius eget enim at varius. Lorem
          ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum pretium
          orci eu semper blandit. Fusce bibendum odio eu dolor elementum,
          accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris lobortis
          semper. Phasellus ut libero quis quam ornare tincidunt quis id diam.
          Pellentesque luctus libero ut dignissim iaculis. Quisque nec orci
          ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus enim,
          eget maximus lorem. Vestibulum ante ipsum primis in faucibus orci
          luctus et ultrices posuere cubilia curae; Pellentesque varius eget
          enim at varius.
        </div>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[13] With overlay with fullcreen panel width and height with restricted panel size", () => {
    const { asFragment } = render(
      <Modal
        options={{
          isOverlay: true,
          isPanelRestrictedSize: true,
          panelWidth: "100%",
          panelHeight: "100%",
        }}
      >
        <div>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum
          pretium orci eu semper blandit. Fusce bibendum odio eu dolor
          elementum, accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris
          lobortis semper. Phasellus ut libero quis quam ornare tincidunt quis
          id diam. Pellentesque luctus libero ut dignissim iaculis. Quisque nec
          orci ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus
          enim, eget maximus lorem. Vestibulum ante ipsum primis in faucibus
          orci luctus et ultrices posuere cubilia curae; Pellentesque varius
          eget enim at varius. Lorem ipsum dolor sit amet, consectetur
          adipiscing elit. Vestibulum pretium orci eu semper blandit. Fusce
          bibendum odio eu dolor elementum, accumsan rhoncus mauris luctus. Sed
          feugiat iaculis mauris lobortis semper. Phasellus ut libero quis quam
          ornare tincidunt quis id diam. Pellentesque luctus libero ut dignissim
          iaculis. Quisque nec orci ullamcorper, pretium quam eu, gravida ipsum.
          Morbi at faucibus enim, eget maximus lorem. Vestibulum ante ipsum
          primis in faucibus orci luctus et ultrices posuere cubilia curae;
          Pellentesque varius eget enim at varius. Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Vestibulum pretium orci eu semper
          blandit. Fusce bibendum odio eu dolor elementum, accumsan rhoncus
          mauris luctus. Sed feugiat iaculis mauris lobortis semper. Phasellus
          ut libero quis quam ornare tincidunt quis id diam. Pellentesque luctus
          libero ut dignissim iaculis. Quisque nec orci ullamcorper, pretium
          quam eu, gravida ipsum. Morbi at faucibus enim, eget maximus lorem.
          Vestibulum ante ipsum primis in faucibus orci luctus et ultrices
          posuere cubilia curae; Pellentesque varius eget enim at varius. Lorem
          ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum pretium
          orci eu semper blandit. Fusce bibendum odio eu dolor elementum,
          accumsan rhoncus mauris luctus. Sed feugiat iaculis mauris lobortis
          semper. Phasellus ut libero quis quam ornare tincidunt quis id diam.
          Pellentesque luctus libero ut dignissim iaculis. Quisque nec orci
          ullamcorper, pretium quam eu, gravida ipsum. Morbi at faucibus enim,
          eget maximus lorem. Vestibulum ante ipsum primis in faucibus orci
          luctus et ultrices posuere cubilia curae; Pellentesque varius eget
          enim at varius.
        </div>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("[14] Without close button", () => {
    const { asFragment } = render(
      <Modal
        options={{
          isPanelCloseButton: false,
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
