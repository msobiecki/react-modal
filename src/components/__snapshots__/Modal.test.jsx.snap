// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MODAL] [1] Initial 1`] = `<DocumentFragment />`;

exports[`[MODAL] [2] Custom duration 1`] = `<DocumentFragment />`;

exports[`[MODAL] [3] Custom z index 1`] = `<DocumentFragment />`;

exports[`[MODAL] [4] With overlay 1`] = `<DocumentFragment />`;

exports[`[MODAL] [5] With custom overlay 1`] = `<DocumentFragment />`;

exports[`[MODAL] [6] With prevent close overlay 1`] = `<DocumentFragment />`;

exports[`[MODAL] [7] With custom panel 1`] = `<DocumentFragment />`;

exports[`[MODAL] [8] With restricted children width 1`] = `<DocumentFragment />`;

exports[`[MODAL] [9] With overlay with restricted children width 1`] = `<DocumentFragment />`;

exports[`[MODAL] [10] With restricted children width with restricted panel size 1`] = `<DocumentFragment />`;

exports[`[MODAL] [11] With overlay with restricted children width with restricted panel size 1`] = `<DocumentFragment />`;

exports[`[MODAL] [12] With fullscreen panel width and height with restricted panel size 1`] = `<DocumentFragment />`;

exports[`[MODAL] [13] With overlay with fullcreen panel width and height with restricted panel size 1`] = `<DocumentFragment />`;

exports[`[MODAL] [14] Without close button 1`] = `<DocumentFragment />`;
